src/main/antlr4/GobraParser.g4

```diff
@@ -155,6 +155,15 @@ assertion:
 
 blockWithBodyParameterInfo: L_CURLY (SHARE identifierList eos)? statementList? R_CURLY;
 
+// Closures
+closureSpecInstance: (qualifiedIdent | IDENTIFIER) (L_CURLY (closureSpecParams COMMA?)? R_CURLY)?;
+
+closureSpecParams: closureSpecParam (COMMA closureSpecParam)*;
+
+closureSpecParam: (IDENTIFIER COLON)? expression;
```

Comment:
Yes, I think it's fine because this always comes after "as" or "implements". Defining it as `qualifiedIdent | IDENTIFIER | compositeLit` might work, but I think it would cause complications (it would allow some invalid code to be parsed, so there would need to be additional checks). Maybe I will try, after resolving the other comments, to see what happens